python/tvm/driver/tvmc/compiler.py

```diff
@@ -0,0 +1,305 @@
+    params : dict
+        The parameters (weights) for the TVM module.
+    dumps : dict
+            Dictionary containing the dumps specified.
+
+    """
+    dump_sources = [x.strip() for x in  dump_sources.split(',')] if dump_sources else None
+    mod, params = frontends.load_model(path, model_format, shapes)
+
+    return compile_relay(
+        mod,
+        params,
+        target,
+        dump_sources=dump_sources,
+        target_host=target_host,
+        tuning_records=tuning_records,
+        alter_layout=alter_layout,
+    )
+
+
+def compile_relay(
+        mod,
+        params,
+        target,
+        dump_sources=None,
+        target_host=None,
+        tuning_records=None,
+        alter_layout=None,
+):
+    """Compile a relay module to a TVM module for the graph runtime.
+
+    Parameters
+    ----------
+    mod : tvm.relay.Module
+        The relay module to compile.
+    params : dict
+        The parameters (weights) for the relay module.
+    target : str
+        The target for which to compile. Can be a plain string or
+        a path.
+    dump_sources : list, optional
+        Dump the generated code for the specified source types, on
+        the requested target.
+    target_host : Union[str, tvm.target.Target], optional
+        The target of the host machine if host-side code
+        needs to be generated.
+    tuning_records: str, optional
+        Name of the file produced by the tuning to be used during
+        compilation.
+    alter_layout: str, optional
+        The layout to convert the graph to. Note, the convert layout
+        pass doesn't currently guarantee the whole of the graph will
+        be converted to the chosen layout.
+
+    Returns
+    -------
+    graph : str
+        A JSON-serialized TVM execution graph.
+    lib : tvm.module.Module
+        A TVM module containing the compiled functions.
+    params : dict
+        The parameters (weights) for the TVM module.
+    dumps : dict
+        Dictionary containing the dumps specified.
+
+    """
+
+    if alter_layout:
+        mod = common.convert_graph_layout(mod, alter_layout)
+
+    if os.path.exists(str(target)):
+        with open(target) as target_file:
+            logging.info("using target input from file: %s", target)
+            target = "".join(target_file.readlines())
+
+    # TODO: We don't have an API to collect a list of supported
+    #       targets yet. (@leandron)
+    logging.debug("creating target from input: %s", target)
+    tvm_target = tvm.target.create(target)
+    target_host = target_host or ""
+
+    if tuning_records:
+        logging.debug("tuning records file provided: %s", tuning_records)
+        with autotvm.apply_history_best(tuning_records):
+            with tvm.transform.PassContext(opt_level=3):
+                logging.debug("building relay graph with tuning records")
+                graph_module = relay.build(mod, tvm_target, params=params, target_host=tvm_target)
```

Comment:
We may need to provide an option to enable graph tuner. If it is not a plan in this PR, add a TODO.

Reply:
It will be introduce in the next PR, related to `tune` subcommand, in parallel with the `run` PR.

I added a TODO.